File: Backend/Models/Enums.cs
namespace PcmBackend.Models
{
    // Hạng thành viên
    public enum MemberTier
    {
        Standard = 0,
        Silver = 1,
        Gold = 2,
        Diamond = 3
    }

    // Loại giao dịch ví
    public enum TransactionType
    {
        Deposit = 0,      // Nạp tiền
        Withdraw = 1,     // Rút tiền
        Payment = 2,      // Thanh toán phí
        Refund = 3,       // Hoàn tiền
        Reward = 4        // Thưởng giải
    }

    // Trạng thái giao dịch
    public enum TransactionStatus
    {
        Pending = 0,      // Chờ duyệt
        Completed = 1,    // Hoàn thành
        Rejected = 2,     // Từ chối
        Failed = 3        // Thất bại
    }

    // Trạng thái booking
    public enum BookingStatus
    {
        PendingPayment = 0,  // Chờ thanh toán
        Confirmed = 1,       // Đã xác nhận
        Cancelled = 2,       // Đã hủy
        Completed = 3        // Hoàn thành
    }

    // Thể thức giải đấu
    public enum TournamentFormat
    {
        RoundRobin = 0,   // Vòng tròn
        Knockout = 1,     // Loại trực tiếp
        Hybrid = 2        // Kết hợp
    }

    // Trạng thái giải đấu
    public enum TournamentStatus
    {
        Open = 0,            // Mở đăng ký
        Registering = 1,     // Đang đăng ký
        DrawCompleted = 2,   // Đã bốc thăm
        Ongoing = 3,         // Đang diễn ra
        Finished = 4         // Kết thúc
    }

    // Trạng thái trận đấu
    public enum MatchStatus
    {
        Scheduled = 0,    // Đã lên lịch
        InProgress = 1,   // Đang diễn ra
        Finished = 2      // Kết thúc
    }

    // Đội thắng
    public enum WinningSide
    {
        Team1 = 1,
        Team2 = 2
    }

    // Loại thông báo
    public enum NotificationType
    {
        Info = 0,
        Success = 1,
        Warning = 2,
        Error = 3
    }

    // Loại danh mục giao dịch (thu/chi)
    public enum TransactionCategoryType
    {
        Income = 0,    // Thu
        Expense = 1    // Chi
    }
}

File: Frontend/lib/models/enums.dart
// Enums matching backend
enum MemberTier { standard, silver, gold, diamond }

enum TransactionType { deposit, withdraw, payment, refund, reward }

enum TransactionStatus { pending, completed, rejected, failed }

enum BookingStatus { pendingPayment, confirmed, cancelled, completed }

enum TournamentFormat { roundRobin, knockout, hybrid }

enum TournamentStatus { open, registering, drawCompleted, ongoing, finished }

enum MatchStatus { scheduled, inProgress, finished }

enum WinningSide { team1, team2 }

enum NotificationType { info, success, warning }

// Extension methods for enum display
extension MemberTierExt on MemberTier {
  String get displayName {
    switch (this) {
      case MemberTier.standard:
        return 'Standard';
      case MemberTier.silver:
        return 'Silver';
      case MemberTier.gold:
        return 'Gold';
      case MemberTier.diamond:
        return 'Diamond';
    }
  }

  String get icon {
    switch (this) {
      case MemberTier.standard:
        return '⭐';
      case MemberTier.silver:
        return '🥈';
      case MemberTier.gold:
        return '🥇';
      case MemberTier.diamond:
        return '💎';
    }
  }
}

extension TransactionTypeExt on TransactionType {
  String get displayName {
    switch (this) {
      case TransactionType.deposit:
        return 'Nạp tiền';
      case TransactionType.withdraw:
        return 'Rút tiền';
      case TransactionType.payment:
        return 'Thanh toán';
      case TransactionType.refund:
        return 'Hoàn tiền';
      case TransactionType.reward:
        return 'Thưởng giải';
    }
  }
}

extension BookingStatusExt on BookingStatus {
  String get displayName {
    switch (this) {
      case BookingStatus.pendingPayment:
        return 'Chờ thanh toán';
      case BookingStatus.confirmed:
        return 'Đã xác nhận';
      case BookingStatus.cancelled:
        return 'Đã hủy';
      case BookingStatus.completed:
        return 'Hoàn thành';
    }
  }
}

extension TournamentStatusExt on TournamentStatus {
  String get displayName {
    switch (this) {
      case TournamentStatus.open:
        return 'Mở đăng ký';
      case TournamentStatus.registering:
        return 'Đang đăng ký';
      case TournamentStatus.drawCompleted:
        return 'Đã bốc thăm';
      case TournamentStatus.ongoing:
        return 'Đang diễn ra';
      case TournamentStatus.finished:
        return 'Kết thúc';
    }
  }
}

extension MatchStatusExt on MatchStatus {
  String get displayName {
    switch (this) {
      case MatchStatus.scheduled:
        return 'Đã lên lịch';
      case MatchStatus.inProgress:
        return 'Đang diễn ra';
      case MatchStatus.finished:
        return 'Kết thúc';
    }
  }
}
